api/src/main/java/org/pmiops/workbench/workspaces/WorkspaceServiceImpl.java

```diff
@@ -394,8 +399,16 @@ public Workspace saveAndCloneCohortsAndConceptSets(Workspace from, Workspace to)
   @Override
   public WorkspaceAccessLevel getWorkspaceAccessLevel(
       String workspaceNamespace, String workspaceId) {
-    String userAccess =
-        fireCloudService.getWorkspace(workspaceNamespace, workspaceId).getAccessLevel();
+    WorkspaceACL workspaceACL = fireCloudService.getWorkspaceAcl(workspaceNamespace, workspaceId);
+    WorkspaceAccessEntry workspaceAccessEntry =
+        Optional.of(workspaceACL.getAcl().get(userProvider.get().getEmail()))
+            .orElseThrow(
+                () ->
+                    new NotFoundException(
+                        String.format(
+                            "Workspace %s/%s not found", workspaceNamespace, workspaceId)));
```

Comment:
It'd help to have a utility function for formatting a fully qualified workspace.

Reply:
I think this is all over the place. Might be worth a tech debt cleanup.

Reply:
https://precisionmedicineinitiative.atlassian.net/secure/RapidBoard.jspa?rapidView=30&projectKey=RW&view=planning&selectedIssue=RW-3365&epics=visible&selectedEpic=RW-1801